src/test/java/com/endava/internship/collections/StudentTest.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    @Test
+    public void whenTheNameOfOneStudentIsGreaterThanTheOther_CompareToShouldReturnPositiveNumber() {
+        Student alex = new Student("Alex", LocalDate.of(2001, Month.JANUARY, 11), "student utm");
+        Student marcel = new Student("Marcel", LocalDate.of(2001, Month.JANUARY, 11), "student ulim");
+
+        Assertions.assertTrue(marcel.compareTo(alex) > 0);
+    }
+
+    @Test
+    public void whenTheNameOfOneStudentIsLessThanTheOther_CompareToShouldReturnNegativeNumber() {
+        Student alex = new Student("Alex", LocalDate.of(2001, Month.JANUARY, 11), "student utm");
+        Student marcel = new Student("Marcel", LocalDate.of(2001, Month.JANUARY, 11), "student ulim");
+
+        Assertions.assertTrue(alex.compareTo(marcel) < 0);
+    }
+
+    @Test
+    public void whenComparingAStudentWithAGreaterDOBThanTheOtherStudentAndEqualNames_CompareToShouldReturnPositiveNumber() {
+        Student alex = new Student("Alex", LocalDate.of(2000, Month.OCTOBER, 18), "student utm");
+        Student alex2 = new Student("Alex", LocalDate.of(2001, Month.JANUARY, 11), "student ulim");
+
+        Assertions.assertTrue(alex2.compareTo(alex) > 0);
+    }
+
+    @Test
+    public void whenComparingAStudentWithDOBLessThanTheOtherStudentAndEqualNames_CompareToShouldReturnNegativeNumber() {
+        Student alex = new Student("Alex", LocalDate.of(2000, Month.OCTOBER, 18), "student utm");
+        Student alex2 = new Student("Alex", LocalDate.of(2001, Month.JANUARY, 11), "student ulim");
+
+        Assertions.assertTrue(alex.compareTo(alex2) < 0);
+    }
+
+    @Test
+    public void getNameShouldReturnTheNameOfTheStudent() {
+        final String ANDREI = "Andrei";
+        Student student = new Student(ANDREI, LocalDate.of(2000, Month.OCTOBER, 18), "student");
+
+        Assertions.assertEquals(ANDREI, student.getName());
+    }
+
+    @Test
+    public void getDateOfBirthShouldReturnTheDateOfBirthOfTheStudent() {
+        final LocalDate DOB = LocalDate.of(2000, Month.OCTOBER, 18);
+        Student student = new Student("Andrei", DOB, "student");
+
+        Assertions.assertEquals(DOB, student.getDateOfBirth());
+    }
+
+    @Test
+    public void getDetailsShouldReturnTheDetailsAboutTheStudent() {
+        final String DETAILS = "student utm";
+        Student student = new Student("Andrei", LocalDate.of(2000, Month.OCTOBER, 18), DETAILS);
+
+        Assertions.assertEquals(DETAILS, student.getDetails());
+    }
+
+    @Test
+    public void toStringShouldReturnAStringContainingTheValuesOfAllTheStudentAttributes() {
+        Student student = new Student("Andrei", LocalDate.of(2000, Month.OCTOBER, 18), "elev bravo");
+
+        String studentString = student.toString();
+        boolean isValid = studentString.contains(student.getName()) &&
+                studentString.contains(student.getDateOfBirth().toString()) &&
+                studentString.contains(student.getDetails());
+
+        Assertions.assertTrue(isValid);
+    }
+}
```

Comment:
As a small remark, please make sure that all files ends with an empty line in order not to have this _**minus sign**_